godot-ffi/src/binding/single_threaded.rs

```diff
@@ -10,13 +10,14 @@
 //! If used from different threads then there will be runtime errors in debug mode and UB in release mode.
 
 use std::cell::Cell;
+use std::thread::ThreadId;
 
 use super::GodotBinding;
 use crate::ManualInitCell;
 
 pub(super) struct BindingStorage {
     // Is used in to check that we've been called from the right thread, so must be thread-safe to access.
-    is_initialized: Cell<bool>,
+    main_thread_id: Cell<Option<ThreadId>>,
```

Comment:
This made me think, do we still want to use the `ThreadId` in release mode, even if it is not used for any validation?

Pro: uses same semantics as Debug, simpler code
Con: release binaries then don't work by default with reloading (but maybe that's good to point to the override?)

Reply:
Release binaries can stop working with reloading accidentally if people just add something that creates a TLS destructor that prevents the reloading anyway in this case. So i dont think that's a huge issue. Maybe we can print out a warning somehow when this potentially has happened?

Reply:
A warning is a good idea 👍

Reply:
It's fine if we do this at a later point too (or not at all), no need to block on this :slightly_smiling_face: